ondemand/README.md

```diff
@@ -685,7 +685,251 @@ Jupyter system app in the menu along with your sandbox development app.
+
+```
+TODO
+```
+
+Notes:
+
+* Passenger detects what app by the presence of a startup file
+* restart the PUN if you change the type of app (ruby => python)
+* see https://www.phusionpassenger.com/ for Passenger documentation
+
+### Status app template
+
+1. My Sandbox Apps. Click New App.
+2. Git Remote: `/var/git/ood-example-ps`.
+3. Launch
+
+#### Benefits for user
+
+#### Benefits for developer
+
+File edit app.rb. Change title. Save. Launch. You can make some changes without app restart.
+
+There is a unit test. You can change the test first, then change the code to verify. Open shell. Execute `rake`.
+
+Notes:
+
+* See tutorial for details: https://osc.github.io/ood-documentation/master/app-development/tutorials-passenger-apps/ps-to-quota.html
+* As an exercise you could change the app to execute  `df --output=target,pcent | tail -n+2`
+* https://github.com/OSC/ood-example-ps
+
+
+### Use the correct environment when developing apps
+
+1. Open Shell app from Sandbox apps. `which rake` and `which ruby`. OnDemand uses SoftwareCollections for RHEL7.
+2. `scl --list` shows the SCLs. To source the environment, `source scl_source enable ondemand`.
+3. For convenience, this was added to .bash_profile: `cat ~/.bash_profile`
+4. OnDemand configured to ssh to OnDemand host for development `cat /etc/ood/config/apps/dashboard/env`
+
+### Apps can use own dependencies
+
+1. Open Shell app to ood-example-ps app
+2. `bundle install --path vendor/bundle`
+3. `touch tmp/restart`
+
+Notes:
+
+- you can use whatever dependencies you want
+- app continues to work even if system libs change
+- app specific dependencies adds a "build" step
+- takes up more space (but space is cheap)
+- very useful during app development to experiment with new packages
+
+
+### NGINX auto serves assets in public/ directory
+
+1. Open Shell app to df app
+2. `mkdir public`
+3. `cp /var/www/ood/apps/sys/jupyter/icon.png public/`
+4. https://localhost:3443/pun/dev/df3/icon.png
+
+Notes:
+
+- when dealing with links to assets or pages in your app, prefix with app suburi
+- app suburi is set in env var `PASSENGER_BASE_URI` set by Passenger
+
+## XDMoD Integration Tutorial
+
+(Optional) submit a job from job composer to demonstrate XDMoD integration with Job Composer:
+
+1. Jobs => Job Composer
+2. Templates
+3. Create New Job (with python template)
+4. Edit Files
+5. Click `jupyter_notebook_data` in tree.
+6. Select `plot_rbm_logistic_classification.py` and click Copy
+7. Go "back" in browser and click Paste
+8. Select script.sh click edit
+9. change `hello.py` to `plot_rbm_logistic_classification.py` and save
+10. Back to Job Composer and submit job
+
+### Enable the integration
+
+Review integration steps (see dashboard MOTD)
+
+1. run command to update config
+2. run command to ingest
+
+Review dashboard widgets - restart Web Server to see
+
+Review Job Composer links - access Job Composer
+
 ## Tutorial Navigation
+
 [Next - Acknowledgments](../docs/acknowledgments.md)
```

Comment:
These lines get messed up if you don't add 2 spaces after them in markdown.  Can you do that so they show up each on their own line?  Thanks!